analysis/src/CompletionFrontEnd.ml

```diff
@@ -1127,6 +1127,45 @@ let completionWithParser1 ~currentFile ~debug ~offset ~path ~posCursor
       (* Case foo-> when the parser adds a ghost expression to the rhs
          so the apply expression does not include the cursor *)
       if setPipeResult ~lhs ~id:"" then setFound ()
+    (*
+       A dot completion for a tagged templated application.
+       Example:
+         sh`echo "meh"`.
+       or
+         sh`bar`.len
+    *)
+    | Pexp_apply
+        {
+          funct = {pexp_desc = Pexp_ident {txt = Lident "."; loc = _}};
+          args =
+            [(_, ({pexp_desc = Pexp_apply _} as innerExpr)); (_, fieldExpr)];
+        }
+      when Res_parsetree_viewer.is_tagged_template_literal innerExpr
+           && expr.pexp_loc |> Loc.hasPos ~pos:posBeforeCursor
+           || CompletionExpressions.isExprHole fieldExpr ->
+      exprToContextPath innerExpr
+      |> Option.iter (fun cpath ->
+             (* Determine the field name if present *)
+             let fieldName =
+               match fieldExpr.pexp_desc with
+               | Pexp_ident {txt = Lident fieldName}
+                 when Res_parsetree_viewer.is_tagged_template_literal innerExpr
+                 ->
+                 fieldName
+               (* This is likely to be an exprhole *)
+               | _ -> ""
```

Comment:
I think we could add the explicit match (or an if branch using the utils we have for expr holes) just to make the code a bit clearer in what we care about in the AST. What do you think?

Reply:
I'm not against having an explicit match, but I don't know what else to return when it is anything else, that is why I went with `| _ -> ""`.

Reply:
Yeah that'd remain, you'd just also add the explicit match for clarity, even though it's redundant.